### PR TITLE
feat: migrate GCP account and update cluster

### PR DESCRIPTION
**Pull request by agaudreault**
This PR migrates the stack the cluster to a new GCP account.

Current account owners: @leoluz and @agaudreault

Changes:

- Add infrastructure as code for the GCP account
- Update several apps
- Remove jenkins app
- Add external-dns to manage the new apps.argoproj.io zone.

Future work

- Some configurations are added manually, mostly secrets (https://github.com/argoproj/argoproj-deployments/issues/29)
  - Use SealedSecrets to manage the secrets as code
- argoproj.io DNS zones need to be migrated to the new GCP account and added to the infra as code. (https://github.com/argoproj/argoproj-deployments/issues/30)
  - Need to move all records outside of apps.argoproj.io, and ultimately infrastructure serving it.
  - The domain needs to be configured accordingly. This will require more sync with the CNCF
- Current admin team config from argoproj is outdated so admin cannot login with the Github auth. (https://github.com/argoproj/argoproj-deployments/issues/31)
  - Need to create/update the admin team in https://github.com/orgs/argoproj/teams.
- Applications are applied manually in the cluster. (https://github.com/argoproj/argoproj-deployments/issues/32)
  - The apps-of-apps pattern could be used
- Infrastructure as code has a user-access module to explicitly define the permissions. This gives permissions to users based on their personal gmail. Some concerns have been raised about having this information in the code. (https://github.com/argoproj/argoproj-deployments/issues/33)
  - Use a data source that will hold the users emails or use google groups 
- Migrate all other resources and delete old project (https://github.com/argoproj/argoproj-deployments/issues/34)
  - Before deleting the old GCP project, all resources needs to be migrated to the infra as code
- CI pipeline to detect terraform change 
- Add more documentation on how to maintain/update the infrastructure and the different components
- Need to update codeowners to review infrastructure changes
  - Create a github team?

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Infrastructure
+/infrastructure/                                      @agaudreault @leoluz
+/argocd/overlays/production/argocd-rbac-cm.yaml       @agaudreault @leoluz
+/external-dns/values.yaml                             @agaudreault @leoluz
```
Comment (leoluz) [inline .github/CODEOWNERS:4]: We should probably remove this for now until we decide how the ownership is going to be defined for the repo.

File: infrastructure/terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Requirements
+
+- OpenTofu
+- GCloud CLI
+- GCP Project
+
+## Bootstrap
+
+To bootstrap the account
```
Comment (leoluz) [inline infrastructure/terraform/README.md:11]: I feel like we are missing some context here. Can you please clarify what bootstrap means in this context, when and how it should be executed?

File: infrastructure/terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## User access
+
+To allow a user to execute the GKE terraform, add it to the [user-access module](./user-access/README.md).
```
Comment (leoluz) [inline infrastructure/terraform/README.md:15]: This link is returning 404 I guess. Can you please doublecheck that this is a valid link?

File: infrastructure/terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+**WARNING:** This gives a lot of permission in the account that are not only restricted to the GKE cluster.
+
+### Argo CD deployment
```
Comment (leoluz) [inline infrastructure/terraform/README.md:19]: I am missing the step by step guide about how to update infrastructure using this solution.

File: infrastructure/terraform/gcp/gke/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# GKE Cluster module
+
+Create GKE clusters
+
+## Known Issues
+
+- Currently, the cluster cannot be update after creation due to https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1876
```
Comment (leoluz) [inline infrastructure/terraform/gcp/gke/README.md:7]: typo: s/update/updaed

File: infrastructure/terraform/gcp/gke/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+## Usage
+
+This modules assume you have the permissions required (given by user-access module) and an existing VPC (created by network module).
+
+To create multiple clusters, add a new file under the `vars/` folder and use the cluster name as the file name.
```
Comment (leoluz) [inline infrastructure/terraform/gcp/gke/README.md:13]: not I understand why someone would want to create multiple clusters in the context of this project. I am missing in this section a step by step procedure describing how to update the infrastructure.

**Review by leoluz — CHANGES_REQUESTED** [review]
Please check my comments.

File: infrastructure/terraform/gcp/gke/README.md (hunk 6 above)
Comment (agaudreault) [reply infrastructure/terraform/gcp/gke/README.md:13]: If something needs to be tested in the future that could impact the current cluster, creating a new one to test and validate changes will be useful and reduce the risk of downtime on the current cluster.

File: infrastructure/terraform/README.md (hunk 2 above)
Comment (agaudreault) [reply infrastructure/terraform/README.md]: I added the steps to apply changes in each module.

File: .github/CODEOWNERS (hunk 1 above)
Comment (agaudreault) [reply .github/CODEOWNERS]: 👍 I commented it for now to keep track of infrastructure related files.

**Review by leoluz — APPROVED** [review]
LGTM

**Comment by tooptoop4** [comment]
@leoluz @agaudreault https://workflows.apps.argoproj.io/workflows/argo still no bueno

**Pull request merged**